test-crawler/main.go

```diff
@@ -100,6 +104,36 @@ func extractPublicMethodsFromFile(ctx context.Context, filePath string) ([]c.Fun
 	return ex.GetExportedFunctions(ctx, filePath)
 }
 
+// makeYAML will make yaml file from public methods.
+func makeYAML(ctx context.Context, filePath string) ([]byte, error) {
+
+	publicMethods, err := extractPublicMethodsFromFile(ctx, filePath)
+	if err != nil {
+		fmt.Print(err)
+		os.Exit(1)
+	}
+
+	var count int
+	for _, fn := range publicMethods {
+		count++
+		fn.ID = count
+		publicMethods = append(publicMethods[1:], fn)
+
```

Comment:
for i := range publicMethods {
  publicMethods[I].Id = count
} 
Isn't this more memory efficient?
Also ID should be autogenerated, not auto incremented

Reply:
`range`  already returns an index: `index, fn`
Therefore, creating a new variable is unnecessary - you should remove `count`.

Reply:
Also, could you explain why we need the `[1:]` in this line:
`publicMethods = append(publicMethods[1:], fn)`

Reply:
Without `[1:]` we will write 2 times methods in yaml file so we need to slice it and not have methods repeat.

Reply:
I see - you are simply modifying the `ID` of each function in the array, correct?
In that case, why not simply modify them in-place? 
```
for i := 0; i < len(publicMethods); i++ {
	publicMethods[i].ID = i
 }
```

Reply:
Yes. Got it. Thanks!